packages/form-state-manager/src/tests/files/use-field.test.js

```diff
@@ -646,4 +646,409 @@ describe('useField', () => {
       expect(managerApi().values.field).toEqual([1, 2, 45]);
     });
   });
+
+  describe('type', () => {
+    let Typper;
+    let wrapper;
+    let spy;
+
+    beforeEach(() => {
+      spy = jest.fn();
+      managerApi = createManagerApi({});
+
+      Typper = ({ spy, ...props }) => {
+        const { input } = useField(props);
+
+        spy(input);
+
+        return <input {...input} />;
+      };
+    });
+
+    it('checkbox with no value', async () => {
+      wrapper = mount(
+        <FormManagerContext.Provider value={{ ...managerApi(), formOptions: managerApi }}>
+          <Typper name="field" type="checkbox" spy={spy} />
+        </FormManagerContext.Provider>
+      );
+
+      expect(spy).toHaveBeenCalledWith({
+        checked: false,
+        multiple: undefined,
+        name: 'field',
+        onBlur: expect.any(Function),
+        onChange: expect.any(Function),
+        onFocus: expect.any(Function),
+        type: 'checkbox',
+        value: undefined
+      });
+      spy.mockReset();
+
+      await act(async () => {
+        wrapper.find('input').simulate('change', { target: { checked: true, type: 'checkbox' } });
+      });
+      wrapper.update();
+
+      expect(spy).toHaveBeenCalledWith({
+        checked: true,
+        multiple: undefined,
+        name: 'field',
+        onBlur: expect.any(Function),
+        onChange: expect.any(Function),
+        onFocus: expect.any(Function),
+        type: 'checkbox',
+        value: undefined
+      });
+    });
+
+    it('checkbox with values', async () => {
```

Comment:
can you add a check that the state value has correct format? For checkbox with no value `values: {foo: true}` and for multiple choice `values: {foo: ['bar', 'baz']}`

Reply:
done